File: src/main/java/com/github/penguin418/todolist/model/request/JoinRequest.java
package com.github.penguin418.todolist.model.request;

import jakarta.validation.constraints.NotNull;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class JoinRequest {
    @NotNull
    private String username;
    @NotNull
    private String password;
    @NotNull
    private String nickname;
}

File: src/main/java/com/github/penguin418/todolist/model/request/JoinRequest.kt
package com.github.penguin418.todolist.model.request

import jakarta.validation.constraints.NotNull

data class JoinRequest(
    @JvmField val username: @NotNull String? = null,
    @JvmField val password: @NotNull String? = null,
    @JvmField val nickname: @NotNull String? = null
)
